cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -354,6 +355,17 @@ void setWorkflowProgramStart(Id.Program programId, String programRunId, String w
                                String workflowNodeId, long startTimeInSeconds, @Nullable String twillRunId);
 
   /**
+   * Logs the stop of the program running under Workflow.
+   * @param workflowRunId workflow run which started the program
+   * @param programRunId program run started by the Workflow
+   * @param stopTimeInSeconds stop time in seconds
+   * @param runStatus final status of the program
+   * @param failureCause cause of failure if the node failed, null otherwise
+   */
+  void setWorkflowProgramStop(ProgramRunId workflowRunId, ProgramRunId programRunId, long stopTimeInSeconds,
+                              ProgramRunStatus runStatus, @Nullable String failureCause);
```

Comment:
What does the failureCause String carries? The message in the `Throwable`? Or something else? The `String` type seems pretty unclear on what is actually there.